pilot/pkg/networking/core/v1alpha3/listener_waypoint.go

```diff
@@ -448,29 +448,19 @@ func (lb *ListenerBuilder) waypointInboundRoute(virtualService config.Config, li
 
 	out := make([]*route.Route, 0, len(vs.Http))
 
-	catchall := false
 	for _, http := range vs.Http {
 		if len(http.Match) == 0 {
 			if r := lb.translateRoute(virtualService, http, nil, listenPort); r != nil {
 				out = append(out, r)
 			}
-			catchall = true
-		} else {
-			for _, match := range http.Match {
-				if r := lb.translateRoute(virtualService, http, match, listenPort); r != nil {
-					out = append(out, r)
-					// This is a catch all path. Routes are matched in order, so we will never go beyond this match
-					// As an optimization, we can just top sending any more routes here.
-					//if isCatchAllMatch(match) {
-					//	catchall = true
-					//	break
-					//}
-				}
-			}
-		}
-		if catchall {
+			// This is a catchall route, so we can stop processing the rest of the routes.
 			break
 		}
+		for _, match := range http.Match {
+			if r := lb.translateRoute(virtualService, http, match, listenPort); r != nil {
+				out = append(out, r)
```

Comment:
should we check catch all , if it it catch all, stop the rest.  To keep consistent with sidecar

Reply:
Not sure. It was like that from the beginning - do not know if it is intentional or a miss. @howardjohn @kyessenov Do you know?

Reply:
This is alpha implement at that time